openapi.yaml

```diff
@@ -0,0 +1,2201 @@
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+    patch:
+      summary: Update a pool
+      operationId: updatePool
+      tags: [Pool]
+      parameters:
+        - $ref: '#/components/parameters/UpdateMask'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Pool'
+
+      responses:
+        '200':
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Connection'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+    delete:
+      summary: Delete a pool
+      operationId: deletePool
+      tags: [Pool]
+      responses:
+        '204':
+          description: No content.
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+
+  /dags/{dag_id}/taskInstances:
+    parameters:
+      - $ref: '#/components/parameters/DAGID'
+      - $ref: '#/components/parameters/FilterExecutionDateStartRange'
+      - $ref: '#/components/parameters/FilterExecutionDateEndRange'
+      - $ref: '#/components/parameters/FilterStartDateStartRange'
+      - $ref: '#/components/parameters/FilterStartDateEndRange'
+      - $ref: '#/components/parameters/FilterEndDateStartRange'
+      - $ref: '#/components/parameters/FilterEndDateEndRange'
+      - $ref: '#/components/parameters/FilterDurationStartRange'
+      - $ref: '#/components/parameters/FilterDurationEndRange'
+      - $ref: '#/components/parameters/FilterState'
+      - $ref: '#/components/parameters/FilterPool'
+      - $ref: '#/components/parameters/FilterQueue'
+    get:
+      summary: Get list of task instance of DAG.
+      description: >
+        This endpoint support reading resources across multiple DAGs by specifying a "-" as a dag_id.
+      operationId: getTaskInstances
+      tags: [Pool]
+      parameters:
+        - $ref: '#/components/parameters/PageLimit'
+        - $ref: '#/components/parameters/PageOffset'
+      responses:
+        '200':
+          description: List of task instances.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/TaskInstanceCollection'
+                  - $ref: '#/components/schemas/CollectionInfo'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+
+  /dags/{dag_id}/taskInstances/{task_id}/{execution_date}:
```

Comment:
How about /dags/dagid/execution_date/taskInstances/...?

Although not explicitly (right now) TIs should be an attribute of the run

Reply:
My thinking here is dag -* dag run -* task instances.

`-*` meaning one to many

Reply:
Here is my response to reading resources across multiple collections.

Reply:
Do you mean to move the TI resource under the DagRun resource?
Before
```
/dags/{dag_id}/dagRuns
/dags/{dag_id}/dagRuns/{execution_date}
/dags/{dag_id}/taskInstances
/dags/{dag_id}/taskInstances/{task_id}/{execution_date}
/dags/{dag_id}/taskInstances/{task_id}/{execution_date}/links
/dags/{dag_id}/taskInstances/{task_id}/{execution_date}/logs/{task_try_number}
/dags/{dag_id}/taskInstances/{task_id}/{execution_date}/xcomEntries
/dags/{dag_id}/taskInstances/{task_id}/{execution_date}/xcomEntries/{key}
```
After:
```
/dags/{dag_id}/dagRuns
/dags/{dag_id}/dagRuns/{execution_date}
/dags/{dag_id}/dagRuns/{execution_date}/taskInstances
/dags/{dag_id}/dagRuns/{execution_date}/taskInstances/{task_id}
/dags/{dag_id}/dagRuns/{execution_date}/taskInstances/{task_id}/links
/dags/{dag_id}/dagRuns/{execution_date}/taskInstances/{task_id}/logs/{task_try_number}
/dags/{dag_id}/dagRuns/{execution_date}/taskInstances/{task_id}/xcomEntries
/dags/{dag_id}/dagRuns/{execution_date}/taskInstances/{task_id}/xcomEntries/{key}
```
That makes sense. We can do it.

Reply:
Yeah

Reply:
Here is commit: https://github.com/apache/airflow/pull/8721/commits/560d15045eba12cf8020680b46325f4ec7f5d912